applications/FreeSurfaceApplication/custom_utilities/edgebased_levelset.h

```diff
@@ -0,0 +1,3738 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics 
+//
+//  License:		 BSD License 
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Antonia Larese
+//
+
+
+#if !defined(KRATOS_EDGEBASED_LEVELSET_FLUID_SOLVER_H_INCLUDED)
+#define  KRATOS_EDGEBASED_LEVELSET_FLUID_SOLVER_H_INCLUDED
+
+//#define SPLIT_OSS
+// #define SYMM_PRESS
+
+
+// System includes
+#include <string>
+#include <iostream>
+#include <algorithm>
+
+// #include <omp.h>
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "includes/deprecated_variables.h"
+#include "includes/node.h"
+#include "includes/cfd_variables.h"
+//#include "geometries/geometry.h"
+#include "utilities/geometry_utilities.h"
+#include "free_surface_application.h"
+
+
+namespace Kratos
+{
+
+template<unsigned int TDim, class MatrixContainer, class TSparseSpace, class TLinearSolver>
+class EdgeBasedLevelSet
+{
+public:
+    //name for the self defined structure
+    typedef EdgesStructureType<TDim> CSR_Tuple;
+    typedef vector<CSR_Tuple> EdgesVectorType;
+
+    //name for row start and column index vectors
+    typedef vector<unsigned int> IndicesVectorType;
+    //defining matrix type for test calculations
+    typedef vector< array_1d<double, TDim> > CalcVectorType;
+    //defining type for local storage of nodal values
+    typedef vector<double> ValuesVectorType;
+
+    //defining types for matrix operations
+    typedef typename TSparseSpace::MatrixType TSystemMatrixType;
+    typedef typename TSparseSpace::VectorType TSystemVectorType;
+
+    typedef std::size_t SizeType;
+
+    //constructor and destructor
+
+    EdgeBasedLevelSet(MatrixContainer& mr_matrix_container,
```

Comment:
Before having the parameters the life was like this!

Reply:
I remember...